contracts/protocol/modules/v1/RebasingComponentModule.sol

```diff
@@ -0,0 +1,281 @@
+
+        _debtIssuanceModule.registerToIssuanceModule(_setToken);
+    }
+
+    /**
+     * @dev MANAGER ONLY: Add rebasing components. Rebasing components are tracked for syncing positions.
+     *
+     * NOTE: ALL ADDED REBASING COMPONENTS CAN BE ADDED AS A POSITION ON THE SET TOKEN WITHOUT MANAGER'S EXPLICIT PERMISSION.
+     * UNWANTED EXTRA POSITIONS CAN BREAK EXTERNAL LOGIC, INCREASE COST OF MINT/REDEEM OF SET TOKEN, AMONG OTHER POTENTIAL UNINTENDED CONSEQUENCES.
+     * SO, PLEASE ADD ONLY THOSE REBASING COMPONENTS WHOSE CORRESPONDING POSITIONS ARE NEEDED AS DEFAULT POSITIONS ON THE SET TOKEN.
+     *
+     * @param _setToken               Instance of the SetToken
+     * @param _newRebasingComponents  Addresses of new rebasing components
+     */
+    function addRebasingComponents(ISetToken _setToken, IERC20[] memory _newRebasingComponents) external onlyManagerAndValidSet(_setToken) {
+        _addRebasingComponents(_setToken, _newRebasingComponents);
+    }
+
+    /**
+     * @dev MANAGER ONLY: Remove rebasing components.
+     * @param _setToken              Instance of the SetToken
+     * @param _rebasingComponents    Addresses of rebasing components to remove
+     */
+    function removeRebasingComponents(ISetToken _setToken, IERC20[] memory _rebasingComponents) external onlyManagerAndValidSet(_setToken) {
+        for(uint256 i = 0; i < _rebasingComponents.length; i++) {
+            IERC20 component = _rebasingComponents[i];
+            require(rebasingComponentEnabled[_setToken][component], "Rebasing component not enabled");
+
+            delete rebasingComponentEnabled[_setToken][component];
+            rebasingComponents[_setToken].removeStorage(address(component));
+        }
+        emit RebasingComponentsUpdated(_setToken, false, _rebasingComponents);
+    }
+
+    /**
+     * @dev MODULE ONLY: Hook called prior to debt issuance module issuance to sync rebasing components position on SetToken.
+     * @param _setToken    Instance of the SetToken
+     */
+    function moduleIssueHook(ISetToken _setToken, uint256 /* _setTokenQuantity */) external override onlyModule(_setToken) {
+        sync(_setToken);
+    }
+
+    /**
+     * @dev MODULE ONLY: Hook called prior to debt issuance module redemption to sync rebasing components position on SetToken.
+     * @param _setToken    Instance of the SetToken
+     */
+    function moduleRedeemHook(ISetToken _setToken, uint256 /* _setTokenQuantity */) external override onlyModule(_setToken) {
+        sync(_setToken);
+    }
+
+    /**
+     * @dev MODULE ONLY: Hook called prior to looping through each component on issuance. Only callable by valid module.
+     */
+    function componentIssueHook(ISetToken _setToken, uint256 /* _setTokenQuantity */, IERC20 /* _component */, bool _isEquity) external override onlyModule(_setToken) {
+        require(_isEquity, "Must be equity");
```

Comment:
this will reinforce the pattern that if there is an external debt positions, then the module that handles it should have an issuance module hook (AV3LM, sync)

Reply:
removed here to ensure issuance isn't unintentionally bricked [`c7b3cdd`](https://github.com/IndexCoop/index-protocol/pull/44/commits/c7b3cdd372b4e2f87e11c934117734137de51378)